docs/administration/configuration/configuring-cluster-ingress/index.md

```diff
@@ -0,0 +1,128 @@
+
+Your custom Ingress must redirect all the requests to `/api` and `/oauth` to the `<helm-release-prefix>-server` Service, while all the rest should be sent to `<helm-release-prefix>-ui`.
+You can find an example of custom Ingress below.
+
+```{ .yaml .copy-to-clipboard }
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: reana-ingress
+  annotations:
+    kubernetes.io/ingress.class: traefik
+    traefik.ingress.kubernetes.io/router.entrypoints: web,websecure
+spec:
+  tls:
+    - secretName: <tls-secret>
+  rules:
+    - http:
+        paths:
+        - path: /api
+          pathType: Prefix
+          backend:
+            service:
+              name: <helm-release-prefix>-server
+              port:
+                number: 80
+        - path: /oauth
+          pathType: Prefix
+          backend:
+            service:
+              name: <helm-release-prefix>-server
+              port:
+                number: 80
+        - path: /
+          pathType: Prefix
+          backend:
+            service:
+              name: <helm-release-prefix>-ui
+              port:
+                number: 80
+      host: <hostname>
+```
+
+## Using reverse proxy
+
+You may be interested in placing a (reverse) proxy service in front of REANA.
+This requires additional configuration to make REANA work correctly.
+
+Let us consider a setup in which HAProxy proxy handles connections to REANA:
+
+```text
+          +-----------+    +---------+
+Client -> |  HAProxy  | -> |  REANA  |
+          +-----------+    +---------+
+```
+
+In this case, REANA has no way to know how the client connected to the server and which protocol the client used, since REANA will only see connections coming from the proxy.
+If HAProxy also performs TLS termination, all the connections to REANA will be carried out over the HTTP protocol, even if the client connected over HTTPS.
+This makes it impossible to construct correct absolute URLs on the server-side, since the traffic protocol is not known.
+
+In these situations, you can make use of a non-standard `X-Forwarded-*` family of HTTP headers to preserve the details about the original connection.
+To fix the particular issue presented before, it is important to configure your proxy to provide correctly the `X-Forwarded-Proto` header.
+
+!!! warning
+
+    Using `X-Forwarded-*` headers can create security issues.
+    You must make sure that the `X-Forwarded-*` headers provided by the client are either discarded or overwritten by a proxy you trust.
+    See for example the [security concerns for X-Forwarded-For](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For#security_and_privacy_concerns).
+
+First of all, the provided Traefik instance must be configured to trust the `X-Forwarded-*` headers coming from HAProxy, so that they won't be overwritten:
```

Comment:
Since the previous text speaks about HAProxy mostly, I would inverse the order and first present what HAProxy configuration is needed, and only then speak of REANA's default proxy amendments.

(In this way the reader basically follows the ASCII art graph from left to right.)

This would give a text like:

You therefore need to configure HAPRoxy to set the `X-Forwarded-Proto` header for each incoming request:

... screenshot

and then configure REANA's provided Traefik ingress to trust the `X-Forwarded-*` headers coming from HAProxy:

... screenshot

Reply:
BTW we could also move the "Warning" box at the end of the main text,  in which case the new section ordering will stand out even clearer.  

We could also place the Warning box simply into the text and instead of "by a proxy you trust" we can directly say "by HAProxy". (And perhaps provide example in the screenshot?)

Reply:
I have kept the warning box as it was before, as I think that it would lose visibility if we move it to the end or if we place it inside the text.

I have also kept the generic _a proxy you trust_ as I wanted this warning to be quite general, as it applies to every proxy (and also to all the applications, not only to REANA). As far as REANA is concerned, only the header `X-Forwarded-Proto` is considered (see [source](https://github.com/reanahub/reana-server/blob/f3bc6327d0580fd4b002211bfdbc97b72e0fc25b/reana_server/config.py#L206)). However, there is no way to tell Traefik to only trust that header, so it would probably be a good idea to overwrite/drop all the other `X-Forwarded-*`  headers, but this might not be needed depending on how the rest of the system is configured...